### PR TITLE
Improve OMR signal code [Unix variant]

### PR DESCRIPTION
**Pull request by babsingh**
Focus is on properly initializing variables.

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

File: port/unix/omrsignal.c [hunk 1]
```diff
@@ -1075,7 +1060,7 @@ masterASynchSignalHandler(int signal, siginfo_t *sigInfo, void *contextInfo)
 static uint32_t
 registerSignalHandlerWithOS(OMRPortLibrary *portLibrary, uint32_t portLibrarySignalNo,  unix_sigaction handler)
 {
-	struct sigaction newAction;
+	struct sigaction newAction = {{0}};
```
Comment (charliegracie) [inline port/unix/omrsignal.c]: Is this the correct way to do this?  I have seen changes recently I believe to resolve issues on newer XCode or Windows compilers that changed these types of initializers to memset calls.

**Comment by charliegracie** [comment]
@genie-omr build all zos

File: port/unix/omrsignal.c (hunk 1 above)
Comment (charliegracie) [reply port/unix/omrsignal.c]: It was for clang compilation on Windows... https://github.com/eclipse/omr/pull/2190/files

Comment (charliegracie) [reply port/unix/omrsignal.c]: I assume this code is compiled via clang on OS X and newer versions of clang will warn about this type of initialization

Comment (babsingh) [reply port/unix/omrsignal.c]: replaced `sigaction` initialization with `memset`.

**Comment by charliegracie** [comment]
@genie-omr build all zos

**Pull request merged**